include/zephyr/drivers/espi_emul.h

```diff
@@ -144,42 +126,61 @@ int espi_emul_register(const struct device *dev, const char *name,
 		       struct espi_emul *emul);
 
 /**
- * Sets the eSPI virtual wire on the host side, which will
- * trigger a proper event(and callbacks) on the emulated eSPI controller
- *
- * @param espi_dev eSPI emulation controller device
- * @param vw The signal to be set.
- * @param level The level of the signal to be set.
+ * Perform the I/O read through the eSPI bus
  *
- * @retval 0 If successful.
- * @retval -EIO General input / output error.
+ * @param dev Device that will handle the operation
+ * @param length number of bytes to read (1/2/4 supported)
+ * @param addr I/O address to read
+ * @param reg Pointer to value which will store the data
+ * @return 0 if IO was successful, -EINVAL if address is not supported
  */
-int emul_espi_host_send_vw(const struct device *espi_dev,
-			   enum espi_vwire_signal vw, uint8_t level);
+int espi_emul_io_read(const struct device *dev,
+	uint8_t length, uint16_t addr, uint32_t *reg);
 
 /**
- * Perform port80 write on the emulated host side, which will
- * trigger a proper event(and callbacks) on the emulated eSPI controller
+ * Perform the I/O write through the eSPI bus
  *
- * @param espi_dev eSPI emulation controller device
- * @param data The date to be written.
+ * @param dev Device that will handle the operation
+ * @param length number of bytes to write (1/2/4 supported)
+ * @param addr I/O address to write
+ * @param reg Data to write
+ * @return 0 if IO was successful, -EINVAL if address is not supported
+ */
+int espi_emul_io_write(const struct device *dev,
+	uint8_t length, uint16_t addr, uint32_t reg);
+
+/**
+ * Set the internal value of virtual-wire on the host side.
  *
- * @retval 0 If successful.
- * @retval -EIO General input / output error.
+ * @param dev Device that will handle the operation
+ * @param vw virtual wire which value will be changed
+ * @param level logic level on virtual wire
+ * @return 0 if IO was successful, -EIO if there's no emulator, -EPERM if
+ *           the direction of vwire is invalid
  */
-int emul_espi_host_port80_write(const struct device *espi_dev, uint32_t data);
+int espi_emul_internal_set_vwire(const struct device *dev,
+	enum espi_vwire_signal vw, uint8_t level);
 
-#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
 /**
- * Get the host device's ACPI shared memory start address. The size of the region is
- * CONFIG_EMUL_ESPI_HOST_ACPI_SHM_REGION_SIZE.
+ * Manage the callback on the host side.
  *
- * @param espi_dev eSPI emulation controller device.
- * @return Address of the start of the ACPI shared memory.
+ * @param dev Device that will handle the operation
+ * @param callback structure describing managed callback
+ * @param set enable or disable the callback
+ * @return 0 if IO was successful, -EIO if there's no emulator, or other values
+ *           returned by the emulator
  */
-uintptr_t emul_espi_host_get_acpi_shm(const struct device *espi_dev);
-#endif
+int espi_emul_manage_host_callback(const struct device *dev,
```

Comment:
"Manage" makes it a bit non-obvious what this does. IMO, `espi_emul_enable_host_callback` would be clearer.

Reply:
It uses the zephyr's generic name of `manage_callback`. All APIs are using this naming for this scheme. Only change here is the `host` part.